crypto/pkcs11key/key.go

```diff
@@ -76,11 +111,15 @@ type Key struct {
 	pin string
 
 	// The public key corresponding to the private key.
-	publicKey rsa.PublicKey
+	rsaPublicKey rsa.PublicKey
```

Comment:
Why not change the type to `crypto.PublicKey`? That way you can avoid having two separate fields, plus the `keyType`. When you need to know the key type, e.g. in `Sign()` you can do a switch statement over the type.